storage/storage_dest.go

```diff
@@ -532,18 +555,32 @@ func (s *storageImageDestination) queueOrCommit(index int, info addedLayerInfo)
 	return nil
 }
 
+// getDiffIDOrTOCDigest returns the diffID for the specified digest or the digest for the TOC, if known.
+func (s *storageImageDestination) getDiffIDOrTOCDigest(uncompressedDigest digest.Digest) (digest.Digest, bool) {
```

Comment:
It would be better for the TOC digest case to return values which clearly can’t conflict with total digests, maybe by using a format that clearly can’t parse as a `digest.Digest`. Then would be able to can rely on the cryptographic hash properties to avoid unwanted layer chain ID conflicts.

See elsewhere about `addedLayerInfo.individualLayerID`.

Reply:
This was ultimately resolved in #2288.